stat/distmv/normal.go

```diff
@@ -365,22 +366,28 @@ func NormalRandCov(dst, mean []float64, cov mat.Symmetric, src rand.Source) []fl
 	case *mat.Cholesky:
 		dstVec := mat.NewVecDense(n, dst)
 		dstVec.MulVec(cov.RawU().T(), dstVec)
-		floats.Add(dst, mean)
 	case *mat.PivotedCholesky:
 		dstVec := mat.NewVecDense(n, dst)
 		dstVec.MulVec(cov.RawU().T(), dstVec)
 		dstVec.Permute(cov.ColumnPivots(nil), true)
-		floats.Add(dst, mean)
-	// TODO(vladimir-ch): enable this case when EigenSym satisfies Matrix.
-	// case *mat.EigenSym:
+	case *mat.EigenSym:
+		d := cov.RawValues()
+		if d[0] < 0 {
+			panic("distmv: covariance not positive semi-definite")
```

Comment:
This is not nice. And the caller doesn't have much options how to avoid it. They should not modify the slice from RawValues (as documented) even if the smallest eigenvalue is a negative almostzero. So they can only modify the covariance matrix and reattempt the factorization. We could hardcode some ad hoc mechanism with a tolerance to consider tiny negative numbers to be zero but that's probably even uglier.

Reply:
Can we allow a matrix type that implements the method set of `EigenSym` and use that instead of `EigenSym`? This way the user can implement an almost zero filter. If this is OK, it would be an argument for keeping `RawQ` since no-one else is using that (though I don't think that is quite protective enough).

Reply:
Something like this? d617f1934dbf66c764bceb2f0c50bbc1e54739f5

Reply:
Yeah, though probably `func NewFilteredEigenSym(ed *mat.EigenSym, epsilon float64) *FilteredEigenSym {`.

Reply:
Initially I had the epsilon but then decided against it. numpy use the epsilon only for checking the eigenvalues, for the computation they use `abs(val)`, so if an eigenvalue is -tol, they use tol. I find it strange, because if the concern is rounding errors and these eigenvalues should really be zero, then why use abs(val) instead of max(0, val)?

Is the name `FilteredEigenSym` good? Would `PositivePartEigenSym` be better?

Reply:
Yes, that is what I came to a few minutes after make in the suggestion.

I prefer `PositivePartEigenSym`.

Reply:
> I prefer PositivePartEigenSym.

Done.